qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/PingEventCollectorTask.java

```diff
@@ -0,0 +1,153 @@
+
+				if (lastBufferCleanAt + clearingInterval < System.currentTimeMillis()) {
+
+					lastBufferCleanAt = System.currentTimeMillis();
+					clearBuffer();
+				}
+
+			} catch (final InterruptedException ex) {
+
+				logger.debug("PingEventCollectorTask run intrrupted");
+				interrupted = false;
+
+			}catch (final Throwable ex) {
+
+				logger.debug(ex.getMessage());
+			}
+		}
+	}
+
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	private void clearBuffer() {
+		logger.debug("clearBuffer started...");
+
+		bufferCleaner.clearBuffer();
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private void addEventToBufferElement (final PingEventBufferElement element, final int position, final MeasurementMonitoringEvent event) {
+		logger.debug("addEventToBufferElement started...");
+
+		if(element.getEventlist()[position] != null) {
+			logger.warn(element.getId() + REPLACEING_MEASURMENT_EVENT + element.getEventlist()[position].toString());
+		}
+
+		element.addEvent(position, event);
+
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private void putEventToBuffer(final EventDTO event) {
+		logger.debug("putEventToBuffer started...");
+
+		final UUID id = UUID.fromString(event.getMetaData().get(QosMonitorConstants.PROCESS_ID_KEY));
+
+		PingEventBufferElement element = eventBuffer.get(id);
+		if (element == null) {
+			element = new PingEventBufferElement(id);
+		}
+
+		try {
+			final QosMonitorEventType eventType = QosMonitorEventType.valueOf(event.getEventType());
+
+			switch (eventType) {
+			case RECEIVED_MONITORING_REQUEST:
+				final ReceivedMonitoringRequestEventDTO validReceivedRequestEvent = EventDTOConverter.convertToReceivedMonitoringRequestEvent(event);
+
+				addEventToBufferElement(element, QosMonitorConstants.RECEIVED_MONITORING_REQUEST_EVENT_POSITION, validReceivedRequestEvent);
+				break;
+			case STARTED_MONITORING_MEASUREMENT:
+				final StartedMonitoringMeasurementEventDTO validStartedEvent = EventDTOConverter.convertToStartedMonitoringMeasurementEvent(event);
+
+				addEventToBufferElement(element, QosMonitorConstants.STARTED_MONITORING_MEASUREMENT_EVENT_POSITION, validStartedEvent);
+				break;
+			case FINISHED_MONITORING_MEASUREMENT:
+				final FinishedMonitoringMeasurementEventDTO validFinishEvent = EventDTOConverter.convertToFinishedMonitoringMeasurementEvent(event);
+
+				addEventToBufferElement(element, QosMonitorConstants.FINISHED_MONITORING_MEASUREMENT_EVENT_POSITION, validFinishEvent);
+				break;
+			case INTERUPTED_MONITORING_MEASUREMENT:
+				final InterruptedMonitoringMeasurementEventDTO validInteruptEvent = EventDTOConverter.convertToInteruptedMonitoringMeasurementEvent(event);
```

Comment:
typo in local variable name. interupt => interrupt

Reply:
fixed